File: src/compiled_language.rs
pub mod compiled_language {
    use crate::language::language::{Language};
    use crate::program_output::program_output::ProgramOutput;

    pub trait CompiledLanguage: Language {
        fn get_executable_path(&self) -> &str;
        fn compile(&self, code_path: &str) -> Result<ProgramOutput, String>;
    }
}

File: src/compiled_language.py
"""
Module containing the interface for compiled languages
"""
from abc import abstractmethod
from subprocess import CompletedProcess

from src.language import Language
from src.result import Result


class CompiledLanguage(Language):
    """
    Interface for compiled languages
    """
    @abstractmethod
    def compile(self, code_path: str) -> Result[CompletedProcess, str]:
        """
        Compiles the code at the given path

        Returns:
            Result[ProgramOutput, str]: The result of the compilation
        """

    @abstractmethod
    def get_executalbe_path(self) -> str:
        """
        Returns the path to the executable
        """
